server/operator.go

```diff
@@ -160,9 +255,28 @@ func (op *regionOperator) GetResourceKind() ResourceKind {
 	return op.Kind
 }
 
+func (op *regionOperator) GetState() OperatorState {
+	return op.State
+}
+
+func (op *regionOperator) SetState(state OperatorState) {
+	if op.State == OperatorFinished || op.State == OperatorTimeOut {
```

Comment:
yes, if one operator combined by multi-operator, some operator in it may be finished or timeout, and now we replace this operator.